### PR TITLE
upgrade deps

### PR DESCRIPTION
**Pull request by stephenmoloney**
## Deps upgrade

- Upgrade dependencies

- One noticeable change is for `Plug ~> 1.5` the `except: [request_url: 1]` to avoid duplication import of the same function from `Plug.Conn`.

- Removed credo dependency on request by @doomspork

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -19,16 +19,15 @@ defmodule Ueberauth.Mixfile do
...
   defp deps do
-    [{:plug, "~> 1.2"},
+    [{:plug, "~> 1.5.0"},
 
-     # dev/test dependencies
-     {:credo, "~> 0.8.10", only: [:dev, :test]},
-     {:earmark, "~> 0.2", only: :dev},
-     {:ex_doc, "~> 0.12", only: :dev}]
+      # dev/test dependencies
+      {:credo, "~> 0.9.0", only: [:dev, :test]},
```
Comment (doomspork) [inline mix.exs]: Let's drop credo since we have the formatter now 👍

**Comment by yordis** [comment]
Credo isn’t just for formatting. It defines an standard way to code as well with good practices.

**Comment by stephenmoloney** [comment]
@doomspork, I'd tend to agree with @yordis about credo - The elixir formatter has a more limited and slighty different feature set whereas credo is a linting tool and a code suggestion/quality tool.
Let me know what you decide though - I'm happy to remove it if you think it best ?

File: mix.exs [hunk 2]
```diff
@@ -19,16 +19,15 @@ defmodule Ueberauth.Mixfile do
   end
 
   def application do
-    [applications: [:logger]]
+    [extra_applications: [:logger]]
```
Comment (yordis) [inline mix.exs]: If I am not mistaking, by doing this, we officially support only `1.4+` ? Or whichever changed the `application` config, I forgot already 😅

Comment (stephenmoloney) [reply mix.exs]: I think that is right - I chnaged it when I noticed that the elixir version in the `mix.exs` file is specified as `elixir: "~> 1.4"` .. I wonder is it already only supporting 1.4 + or is that a discretionary keyword.. not sure. I can change this back if needed?

File: mix.exs (hunk 1 above)
Comment (stephenmoloney) [reply mix.exs]: credo is dropped. Thanks for the review. :+1:

**Pull request merged**